### PR TITLE
Edgectl Intercept using mappings

### PR DESCRIPTION
**Pull request by kflynn**
Neuter a lot of the old sidecar stuff and use an actual Ambassador to do routing, with `edgectl intercept` managing the mappings for the intercepts.

It's probably simplest to review the whole thing at once, rather than commit-by-commit.

### CODE REVIEW

File: python/ambassador/ir/ir.py [hunk 1]
```diff
@@ -426,12 +429,19 @@ def agent_finalize(self, aconf) -> None:
         # routed insecure. !*@&#*!@&#* We need per-mapping security settings.
         #
         # XXX What if they already have a mapping with this name?
+
+        ctx_name = None
+
+        if self.agent_orgination_ctx:
+            ctx_name = self.agent_origination_ctx.name
```
Comment (ark3) [inline python/ambassador/ir/ir.py]: These are spelled differently. Or my eyes are _really_ bad today.

File: python/ambassador/ir/ir.py [hunk 2]
```diff
@@ -417,14 +421,18 @@ def agent_finalize(self, aconf) -> None:
 
         self.logger.info(f"Intercept agent active for {self.agent_service}:{agent_port}, adding fallback mapping")
 
+        # XXX OMG this is a crock. Don't use precedence -1000000 for this, because otherwise Edge
+        # Control might decide it's the Edge Policy Console fallback mapping and force it to be
+        # routed insecure. !*@&#*!@&#* We need per-mapping security settings.
```
Comment (ark3) [inline python/ambassador/ir/ir.py]: "Edge Control" -> "Edge Stack"?

File: python/ambassador/ir/ir.py (hunk 1 above)
Comment (kflynn) [reply python/ambassador/ir/ir.py]: You were right, as it happens. FFS.

File: python/ambassador/ir/ir.py (hunk 2 above)
Comment (kflynn) [reply python/ambassador/ir/ir.py]: Done.

**Pull request merged**